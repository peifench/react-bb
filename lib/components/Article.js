import React from 'react';

const styles = {
  article: { padding: 10 },
};

const dateDisplay = (string) =>
  new Date(string).toDateString();

const Article = (props) => {
  const { article, actions } = props;
  const author = actions.lookupAuthor(article.authorId);
  return (
    <div style={styles.article}>
      <div>{article.title}</div>
      <div>{dateDisplay(article.date)}</div>
      <div>
        <a href={author.website}>
          {author.firstName} {author.lastName}
        </a>
      </div>
      <div>{article.body}</div>
    </div>
  );
};

export default Article;
