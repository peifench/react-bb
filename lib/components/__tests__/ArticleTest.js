import React from 'react';
import { Article } from '../Article';
import renderer from 'react-test-renderer';

describe('Article', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <Article article={{}} author={{}} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
