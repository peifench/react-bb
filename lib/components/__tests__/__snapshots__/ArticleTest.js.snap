// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<div
  style={
    Object {
      "padding": 10,
    }
  }
>
  <div />
  <div>
    Invalid Date
  </div>
  <p>
    <a
      href={undefined}
    >
       
    </a>
  </p>
  <div />
</div>
`;
